python/mxnet/kvstore.py

```diff
@@ -167,30 +162,29 @@ def push(self, key, value, priority=0):
             ctypes.c_int(priority)))
 
     def pull(self, key, out=None, priority=0):
-        """ Pull a single value or a sequence of values from the store.
-
-        Data consistency:
+        """ Pulls a single value or a sequence of values from the store.
 
-        1. This function returns after adding an operator to the engine. But any
-        further read on out will be blocked until it is finished.
+        This function returns immediately after adding an operator to the engine.
+        Subsequent attempts to read from the `out` variable will be blocked until the
+        pull operation completes.
 
-        2. `pull` is always called after all previous push and pull on the same
-        key are finished.
+        `pull` is executed asynchronously after all previous `push` and `pull` calls
+        for the same input key(s) are finished.
 
-        3. It pulls the newest value from the store.
+        The returned values are gauranteed to the latest values in the store.
```

Comment:
"gauranteed to the latest values in the store" what does this mean? shouldn't it be "guaranteed to be..."?